File: Code/Tamagotchi.Common/Models/AnimalModel.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Tamagotchi.Common.Models
{
    public class AnimalModel : BaseModel
    {
        [Required]
        public string Name { get; set; }
        [Required]
        public string Description { get; set; }

        public int TimesDownloaded { get; set; }
        public bool IsReady { get; set; }
        public bool IsActive { get; set; }

        [Required]
        public string IdleUri { get; set; }
        [Required]
        public string PlayUri { get; set; }
        [Required]
        public string EatUri { get; set; }
        [Required]
        public string SleepUri { get; set; }
        [Required]
        public string PacketUri { get; set; }
        
        [Required]
        public string PacketFile { get; set; }
        [Required]
        public string PlayImage { get; set; }
        [Required]
        public string EatImage { get; set; }
        [Required]
        public string SleepImage { get; set; }
        [Required]
        public string IdleImage { get; set; }

        public UserModel User { get; set; }
        public IEnumerable<LogModel> Logs { get; set; }
    }
}

File: Code/Tamagotchi.Presentation/Client.Website/tamagotchi-web-angular/src/app/shared/models/AnimalModel.ts
export interface AnimalModel {
    Name: string;
    Description: string;
    TimesDownloaded: number;
    IsReady: Boolean;
    IsActive: Boolean;
    IdleUri: string;
    PlayUri: string;
    EatUri: string;
    SleepUri: string;
    PacketUri: string;
    UserId: string;
    AnimalId: string;
    DateCreated: string;
    LastModified: string;
}

//TODO: Datatypes?
//In table details: creator or owner, message of this pet or just the last animal.